### ISSUE TITLE
JS tokenizer incorrect for inline conditionally created immediately invoked anon function

### ISSUE THREAD

**Issue by gsherwood** [squizlabs/PHP_CodeSniffer#382]
Running the Squiz standard over this JS code:

``` js
if (window.jQuery)(function($) {
    $.fn.reset = function() {
        return this.each(function() {
            try {
                this.reset();
            } catch (e) {}
        });
    };
})(jQuery);
```

Causes this error: 

```
Notice: Undefined index: scope_closer in CodeSniffer/Standards/Squiz/Sniffs/ControlStructures/ControlSignatureSniff.php on line 189
```

Removing the parenthesis, or adding braces to the IF statement allows tokenizing to work correctly.
